BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigRefresherTest.cs

```diff
@@ -74,6 +75,50 @@ public void SelfDiagnosticsConfigRefresher_CaptureAsConfigured()
             }
         }
 
+        [TestMethod]
+        public void SelfDiagnosticsConfigRefresher_ReadFromEnviornmentVars_OnlyLogDirectoryWasSet()
+        {
+            Environment.SetEnvironmentVariable("LogDirectory", ".");
+
+            using (var configRefresher = new SelfDiagnosticsConfigRefresher())
+            {
+                // Emitting event of EventLevel.Error
+                CoreEventSource.Log.InvalidOperationToStopError();
+                var filePath = configRefresher.CurrentFilePath;
+
+                int bufferSize = 512;
+                byte[] actualBytes = ReadFile(filePath, bufferSize);
+                string logText = Encoding.UTF8.GetString(actualBytes);
+                Assert.IsTrue(logText.StartsWith(MessageOnNewFileString));
+            }
+        }
+
+        [TestMethod]
+        public void SelfDiagnosticsConfigRefresher_ReadFromEnviornmentVars()
```

Comment:
this test could be flaky.
tests often run in parallel, and this could capture a different message.

Reply:
I've added logic to clear the environment variable and force initialize of PlatformSingleton to address the concern. Thanks!